packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -206,14 +207,13 @@ abstract class MultiChildLayoutDelegate {
     return DiagnosticsProperty<RenderBox>('${childParentData.id}', child);
   }
 
-  void _callPerformLayout(Size size, RenderBox firstChild) {
+  void _callPerformLayout(Size size, RenderBox? firstChild) {
     // A particular layout delegate could be called reentrantly, e.g. if it used
     // by both a parent and a child. So, we must restore the _idToChild map when
     // we return.
-    final Map<Object, RenderBox> previousIdToChild = _idToChild;
+    final Map<Object, RenderBox>? previousIdToChild = _idToChild;
```

Comment:
this can be non-nullable if _idToChild ends up non-nullable

Reply:
same as _debugChildrenNeedingLayout

Reply:
I'm pretty sure we can make _idToChild non-nullable. I think it's how I had it working. Might require a bit of a tweak to the logic?